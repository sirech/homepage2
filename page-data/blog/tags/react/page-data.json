{
    "componentChunkName": "component---src-templates-blog-index-js",
    "path": "/blog/tags/react",
    "result": {"pageContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"},"group":[{"frontmatter":{"layout":"post","title":"Applications Instead of Libraries","path":"/applications-instead-of-libraries/","categories":["Architecture","React","Micro frontends","JavaScript"],"date":"2021/11/20","draft":false,"description":"Using module federation to implement micro frontends"}},{"frontmatter":{"layout":"post","title":"From Redux to the Context API: A Practical Migration Guide","path":"/redux-to-context-practical-migration-guide/","categories":["React","Redux","Context API","Architecture","State Management"],"date":"2021/06/14","draft":false,"description":"Let's migrate a React application built with Redux to the Context API and see how it looks like"}},{"frontmatter":{"layout":"post","title":"Starting you new React application with Redux? Consider a simpler approach using the Context API","path":"/starting-a-new-app-with-redux-consider-context-api-first/","categories":["React","Redux","Context API","Architecture"],"date":"2021/05/11","draft":false,"description":"Traditionally, starting a new React application means adding Redux for state management. Let's talk about the Context API as an alternative"}},{"frontmatter":{"layout":"post","title":"Authorization Code Flow with PKCE (OAuth) in a React application","path":"/oauth-authorization-code-flow-pkce-for-react/","categories":["Auth0","OAuth","React"],"date":"2021/03/29","draft":false,"description":"SPAs implementing OAuth should pick Authorization Code Flow with PKCE for maximum security. Let's implement it in React with help of Auth0"}},{"frontmatter":{"layout":"post","title":"A directory structure for React projects","path":"/a-directory-structure-for-react-projects/","categories":["React","TypeScript","Code Organization"],"date":"2020/07/19","draft":false,"description":"How do you structure your components in a React application? I like to group mine around features close to the domain enforced by convention"}},{"frontmatter":{"layout":"post","title":"Fail a test in Jest if an unexpected network request happens","path":"/jest-fail-test-if-unexpected-network-request-happens/","categories":["React","Jest","Testing","react-testing-library","create-react-app","axios"],"date":"2019/12/08","draft":false,"description":"Automatically failing unit tests with Jest in React when network requests are made will make your tests more reliable and easier to maintain"}},{"frontmatter":{"layout":"post","title":"Custom Components in Formik","path":"/custom-components-in-formik/","categories":["React","Formik","react-hooks"],"date":"2019/08/15","draft":false,"description":"Formik is the new cool kid on the block for forms in react. Here is how to build custom input elements on top"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby 2","path":"/migrating-to-gatsby2/","categories":["React","Gatsby"],"date":"2018/10/03","draft":false,"description":"Gatsby 2 is out! How did the migration go?"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 3","path":"/angular-from-react-part3/","categories":["JavaScript","React","Angular","TypeScript","angular-cli","redux"],"date":"2018/07/07","draft":false,"description":"Third part of my series on Angular, coming from a React background. This article focus on state management"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 2","path":"/angular-from-react-part2/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/04/29","draft":false,"description":"Second part of my series on Angular, coming from a React background. This article focus on testing"}},{"frontmatter":{"layout":"post","title":"Angular from the perspective of a React fan - Part 1","path":"/angular-from-react-part1/","categories":["JavaScript","React","Angular","TypeScript","angular-cli"],"date":"2018/03/25","draft":false,"description":"First part of my series on Angular, coming from a React background. This article is about bootstrapping and TypeScript"}},{"frontmatter":{"layout":"post","title":"Migrating to Gatsby","path":"/migrating-to-gatsby/","categories":["React","Gatsby"],"date":"2018/03/18","draft":false,"description":"It all starts here. After using Create React App for a while, I decided to switch to Gatsby for my personal website"}}],"pathPrefix":"blog/tags/react","first":true,"last":true,"index":1,"pageCount":1,"additionalContext":{"siteMetadata":{"title":"Mario Fernandez","description":"This is the blog from Mario Fernandez, a Software Developer working for ThoughtWorks","author":"Mario Fernandez","url":"https://hceris.com","twitter":"@sirech"}}}},
    "staticQueryHashes": []}