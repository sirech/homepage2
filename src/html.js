import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'

let stylesStr
if (process.env.NODE_ENV === 'production') {
  try {
    // eslint-disable-next-line
    stylesStr = require('!raw-loader!../public/styles.css')
  } catch (e) {
    console.log(e)
  }
}

class HTML extends React.Component {
  render() {
    const head = Helmet.rewind()
    let css
    if (process.env.NODE_ENV === 'production') {
      css = (
        <style
          id="gatsby-inlined-css"
          dangerouslySetInnerHTML={{ __html: stylesStr }}
        />
      )
    }

    return (
      <html lang="en">
        <head>
          <meta charSet="utf-8" />
          <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1.0"
          />
          {this.props.headComponents}
          {css}
          <link
            href="/img/apple-touch-icon.png"
            rel="apple-touch-icon"
            sizes="180x180"
          />
          <link href="/img/favicon.ico" rel="icon" type="image/x-icon" />
        </head>
        <body itemScope itemType="http://schema.org/WebPage">
          <div
            id="___gatsby"
            dangerouslySetInnerHTML={{ __html: this.props.body }}
          />
          {this.props.postBodyComponents}
        </body>
      </html>
    )
  }
}

HTML.propTypes = {
  body: PropTypes.string.isRequired,
  headComponents: PropTypes.node,
  postBodyComponents: PropTypes.node,
}

export default HTML
