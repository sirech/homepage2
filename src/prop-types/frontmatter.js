import PropTypes from 'prop-types'

export default PropTypes.shape({
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  date: PropTypes.string.isRequired,
  layout: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  draft: PropTypes.bool.isRequired,
  description: PropTypes.string,
}).isRequired
