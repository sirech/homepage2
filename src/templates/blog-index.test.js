import React from 'react'
import { render } from '@testing-library/react'

import IndexPage from './blog-index'

import frontmatter from '../fixtures/frontmatter'
import siteMetadata from '../fixtures/siteMetadata'
import html from '../fixtures/html'

describe('components', () => {
  describe('IndexPage', () => {
    it('renders correctly', () => {
      const site = {
        siteMetadata: siteMetadata(),
      }

      const post = { id: 'id', frontmatter: frontmatter(), html: html() }

      const pageContext = {
        index: 3,
        pageCount: 5,
        additionalContext: site,
        group: [{ post }, { post }],
      }

      const { asFragment } = render(<IndexPage pageContext={pageContext} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
