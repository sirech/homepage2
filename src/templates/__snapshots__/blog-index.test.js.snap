// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components IndexPage renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <div
        class="articles"
      >
        <article
          class="article articleBox"
        >
          <header>
            <a
              href="/the-post"
            >
              <h3>
                My Post
              </h3>
            </a>
            <time
              datetime="2018/03/01"
            >
              2018/03/01
            </time>
            <a
              href="/blog/tags/javascript"
            >
              <span
                class="tag badge mr-1"
              >
                JavaScript
              </span>
            </a>
            <a
              href="/blog/tags/react"
            >
              <span
                class="tag badge mr-1"
              >
                React
              </span>
            </a>
          </header>
          <section>
            this is the description
          </section>
        </article>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="articles"
      >
        <article
          class="article articleBox"
        >
          <header>
            <a
              href="/the-post"
            >
              <h3>
                My Post
              </h3>
            </a>
            <time
              datetime="2018/03/01"
            >
              2018/03/01
            </time>
            <a
              href="/blog/tags/javascript"
            >
              <span
                class="tag badge mr-1"
              >
                JavaScript
              </span>
            </a>
            <a
              href="/blog/tags/react"
            >
              <span
                class="tag badge mr-1"
              >
                React
              </span>
            </a>
          </header>
          <section>
            this is the description
          </section>
        </article>
      </div>
    </div>
    <div
      class="container"
    >
      <nav
        class="mt-4"
      >
        <ul
          class="pagination justify-content-center"
        >
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="/blog/2"
            >
              <i
                class="fa fa-caret-left"
              />
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="/blog/"
            >
              1
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="/blog/2"
            >
              2
            </a>
          </li>
          <li
            class="page-item active"
          >
            <a
              class="page-link"
              href="/blog/3"
            >
              3
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="/blog/4"
            >
              4
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="/blog/5"
            >
              5
            </a>
          </li>
          <li
            class="page-item"
          >
            <a
              class="page-link"
              href="/blog/4"
            >
              <i
                class="fa fa-caret-right"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
